samples/drivers/touch/src/main.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2024 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sample, LOG_LEVEL_INF);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/drivers/display.h>
+
+#define CROSS_WIDTH  20
+#define CROSS_HEIGHT 20
+#define BUFFER_SIZE  ((CROSS_WIDTH) * (CROSS_HEIGHT) * 4)
+#define REFRESH_RATE 100
+
+#ifndef DT_CHOSEN_zephyr_touch
+#error "Unsupported board: zephyr,touch is not assigned"
+#endif
+
+#ifndef DT_CHOSEN_zephyr_display
+#error "Unsupported board: zephyr,display is not assigned"
+#endif
+
+static const struct device *const display_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_display));
+static const struct device *const touch_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_touch));
+static struct display_capabilities capabilities;
+static struct display_buffer_descriptor buf_desc = {.buf_size = BUFFER_SIZE,
+						    .pitch = CROSS_WIDTH,
+						    .width = CROSS_WIDTH,
+						    .height = CROSS_HEIGHT};
+
+static uint8_t buffer_cross[BUFFER_SIZE] = {0};
+static const uint8_t buffer_cross_empty[BUFFER_SIZE] = {0};
+static struct k_sem sync_x, sync_y;
+
+static struct {
+	size_t x;
+	size_t y;
+} touch_point, touch_point_drawn;
+
+static void touch_event_callback(struct input_event *evt, void *user_data)
+{
+	if (evt->code == INPUT_ABS_X) {
+		touch_point.x = evt->value;
+		k_sem_give(&sync_x);
+	}
+	if (evt->code == INPUT_ABS_Y) {
+		touch_point.y = evt->value;
+		k_sem_give(&sync_y);
+	}
+}
+
+static void clear_screen(void)
+{
+	int x;
+	int y;
+
+	for (x = 0; x < capabilities.x_resolution; x += CROSS_WIDTH) {
+		for (y = 0; y < capabilities.y_resolution; y += CROSS_HEIGHT) {
+			display_write(display_dev, x, y, &buf_desc, buffer_cross_empty);
+		}
+	}
+}
+
+static void fill_cross_buffer(int bpp)
+{
+	int i;
+	int x;
+	int y;
+	int index;
+
+	for (i = 0; i < bpp; i++) {
+		for (x = 0; x < CROSS_WIDTH; x++) {
+			index = bpp * (CROSS_HEIGHT / 2 * CROSS_WIDTH + x);
+			buffer_cross[index + i] = -1;
+		}
+		for (y = 0; y < CROSS_HEIGHT; y++) {
+			index = bpp * (y * CROSS_WIDTH + CROSS_WIDTH / 2);
+			buffer_cross[index + i] = -1;
+		}
+	}
+}
+
+int main(void)
+{
+	size_t tick_count;
+	size_t bpp;
```

Comment:
If the api does not require it, I would avoid using posix types.

Reply:
Changed